server/server_fuzz.go

```diff
@@ -0,0 +1,87 @@
+// +build gofuzz
+
+package server
+
+import (
+	"bytes"
+	"io"
+	"reflect"
+)
+
+// FuzzLocalStorage tests the Local Storage.
+func FuzzLocalStorage(fuzz []byte) int {
+	if len(fuzz) == 0 {
+		return -1
+	}
+
+	storage, err := NewLocalStorage("/tmp", nil)
+	if err != nil {
+		return 0
+	}
+
+	input := bytes.NewReader(fuzz)
+	err = storage.Put("token", "filename", input, "application/octet-stream", uint64(len(fuzz)))
+	if err != nil {
+		return 0
+	}
+
+	contentType, contentLength, err := storage.Head("token", "filename")
+	if err != nil {
+		return 0
+	}
+
+	if contentType != "application/octet-stream" {
+		return 0
+	}
+
+	if contentLength != uint64(len(fuzz)) {
+		return 0
+	}
+
+	output, contentType, contentLength, err := storage.Get("token", "filename")
+	if err != nil {
+		return 0
+	}
+
+	if contentType != "application/octet-stream" {
+		return 0
+	}
+
+	if contentLength != uint64(len(fuzz)) {
+		return 0
+	}
+
+	var length int
+	b := make([]byte, len(fuzz))
+	for {
+		n, err := output.Read(b)
+		length += n
+		if err == io.EOF {
+			break
+		}
+	}
+
+	if !reflect.DeepEqual(b, fuzz) {
+		return 0
+	}
+
+	if length != len(fuzz) {
+		return 0
+	}
+
+	err = storage.Delete("token", "filename")
+	if err != nil {
+		return 0
+	}
+
+	_, _, err = storage.Head("token", "filename")
+	if err != nil {
```

Comment:
Maybe this err check is not needed. I think it prevents the IsNotExist check from happening.

Reply:
indeed